src/Carbon/Traits/Difference.php

```diff
@@ -311,23 +311,25 @@ public function secondsUntilEndOfDay()
      * 5 months after
      *
      * @param Carbon|null $other
-     * @param bool        $absolute removes time difference modifiers ago, after, etc
-     * @param bool        $short    displays short format of time units
-     * @param int         $parts    displays number of parts in the interval
+     * @param int         $syntax difference modifiers (ago, after, etc) rules
+     * @param bool        $short  displays short format of time units
+     * @param int         $parts  displays number of parts in the interval
      *
      * @return string
      */
-    public function diffForHumans($other = null, $absolute = false, $short = false, $parts = 1)
+    public function diffForHumans($other = null, $syntax = CarbonInterface::DIFF_RELATIVE_AUTO, $short = false, $parts = 1)
     {
         /* @var CarbonInterface $this */
-        $isNow = $other === null;
         $interval = [];
+        $syntax = (int) $syntax;
```

Comment:
This will change result for non-numeric strings or floats less than 1 which become 0 when casted, but previously would evaluate to true.

Reply:
Yes but previously, `2` and `3` was evaluated to `true`, now they are special modes. It's not meant to be a perfect backward compatibility support, `boolean` well formed as in the doc and unit tests will continue to work the same and I think it's enough backward compatibility for a change that comes in a major version upgrade.

Reply:
Compatibility could be stronger with just these two special cases. On the other hand I expect most people typed true so it's probably fine.